pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/DynamicBrokerSelector.java

```diff
@@ -112,7 +112,7 @@ public String selectBroker(String... tableNames) {
 
   @Override
   public List<String> getBrokers() {
-    return _brokerList;
+    return _allBrokerListRef.get();
```

Comment:
Was there a consumer of this method that relied on the list of ZK servers, as returned previously?

Reply:
Shall we return an immutable list of brokers?

Reply:
> Was there a consumer of this method that relied on the list of ZK servers, as returned previously?
There is no consumer in the Pinot code base. I dont know if external clients use it. I verified with @Jackie-Jiang that this is a bug. 

> Shall we return an immutable list of brokers?
I am keeping the PR as small as possible.